src/services/sync_coordinator.py

```
@@ -3,30 +3,32 @@
     """Coordinates synchronization between git repo and vector embeddings."""
 
     def __init__(
         self,
-        repo_url: str,
+        git_manager: GitManagerProtocol,
         local_path: str,
         vector_store_path: str = "./chroma_db",
-        branch: str = "main",
-        github_token: str = "",
         embedding_model: str = "sentence-transformers/all-MiniLM-L6-v2",
+        processing_path: str = None,
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)